packages/flutter/test/widgets/text_test.dart

```diff
@@ -1662,29 +1662,48 @@ void main() {
   });
 
   testWidgets('Mouse hovering over selectable Text uses SystemMouseCursor.text', (WidgetTester tester) async {
-    await tester.pumpWidget(const MaterialApp(
-      home: SelectionArea(
-        child: Text('Flutter'),
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: SelectionArea(
+          child: Align(
+            alignment: Alignment.topLeft,
+            child: SizedBox(
+              width: 300,
+              height: 300,
+              child: Text('Flutter', style: TextStyle(fontSize: 50)),
+            ),
+          ),
+        ),
       ),
-    ));
+    );
 
     final TestGesture gesture = await tester.createGesture(kind: PointerDeviceKind.mouse, pointer: 1);
-    await gesture.addPointer(location: tester.getCenter(find.byType(Text)));
 
-    await tester.pump();
+    expect(tester.getRect(find.byType(Text)), const Rect.fromLTWH(0, 0, 300, 300));
 
+    // Mouse hovering over selectable text.
+    await gesture.addPointer(location: const Offset(10, 10));
+    await tester.pump();
     expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), SystemMouseCursors.text);
+
+    // Mouse is still inside the `Text` widget, but not the actual rendered text.
+    await gesture.moveTo(const Offset(100, 100));
+    await tester.pump();
+    expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), SystemMouseCursors.basic);
+
   });
 
   testWidgets('Mouse hovering over selectable Text uses default selection style mouse cursor', (WidgetTester tester) async {
-    await tester.pumpWidget(MaterialApp(
-      home: SelectionArea(
-        child: DefaultSelectionStyle.merge(
-          mouseCursor: SystemMouseCursors.click,
-          child: const Text('Flutter'),
+    await tester.pumpWidget(
+      MaterialApp(
+        home: SelectionArea(
+          child: DefaultSelectionStyle.merge(
+            mouseCursor: SystemMouseCursors.click,
+            child: const Center(child: Text('Flutter')),
```

Comment:
Why is this `Center` needed?

Reply:
add center so when add point : `await gesture.addPointer(location: tester.getCenter(find.byType(Text)));`

the center of the text widget is within the actual text.